include/ews/ews.hpp

```diff
@@ -7602,6 +7602,545 @@ namespace ews
     static_assert(std::is_move_assignable<internet_message_header>::value, "");
 #endif
 
+    //! The ExtendedFieldURI element identifies an extended MAPI property.
+    class extended_field_uri final
+    {
+    public:
+        //! \brief Represents the <tt>\<PropertySetId\></tt> of an
+        //! <tt>\<ExtendedFieldUri\></tt>.
+        //!
+        //! Defines the well-known property set IDs for extended MAPI
+        //! properties. If this attribute is used, the PropertySetId and
+        //! PropertyTag attributes cannot be used. This attribute must be
+        //! used with either the PropertyId or PropertyName attribute, and
+        //! the PropertyType attribute. The DistinguishedPropertySetId
+        //! Attribute table later in this topic lists the possible values
+        //! for this attribute. This attribute is optional.
+        class distinguished_property_set_id final
+        {
+        public:
+//! Default constructor
+#ifdef EWS_HAS_DEFAULT_AND_DELETE
+            distinguished_property_set_id() = default;
+#else
+            distinguished_property_set_id(){};
+#endif
+            //! Constructor to create an DistinguishedPropertySetId property
+            distinguished_property_set_id(std::string str)
+                : _str(std::move(str)){};
```

Comment:
This ctor should be a non-converting constructor. (Hint: Use the explicit specifier.)

Because this is the whole point of the class: to use C++'s type system to distinguish all those different strings that can be used as attributes of `<ExtendedFieldURI/>`. To catch errors at compile-time rather than run-time.